### PR TITLE
Config: Add $include which replaces $data and can include any value

### PR DESCRIPTION
**Pull request by jhaals**
## Hey, I just made a Pull Request!

This PR adds a new `$include` keyword to the config parser which replaces `$data` and allows for json objects and other structured data to be loaded in from external files.

We are planning to use this for the Github Apps integration which is going to require a collection of multiple secrets.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)

### CODE REVIEW

**Review by benjdlambert — APPROVED** [review]

**Comment by benjdlambert** [comment]
Welcome :tada: :rocket:

File: packages/config-loader/src/lib/secrets.ts [hunk 1]
```diff
@@ -134,6 +140,35 @@ export async function readSecret(
 
     return String(value);
   }
+  if ('include' in secret) {
+    const [filePath, dataPath] = secret.include.split(/#(.*)/);
```
Comment (benjdlambert) [inline packages/config-loader/src/lib/secrets.ts:144]: Build says that `secret.include` is never and it's failing 🤷 

Wonder if we need to add `include` to the `secret` type? not sure where that comes from/.

**Comment by freben** [comment]
"include" is a verb, where all the others are nouns. The verb won't let us discern between different parsing types, such as rhe difference between "include this json file as a text blob" vs "include the structured contents of this (part of?) json file". Just noting, not meant as a rejecting review, wasn't part of the discussions :)

**Comment by Rugvip** [comment]
@freben The intention is "include as if it was part of the original config file". We're not processing it atm, but will likely add that in the future

**Review by Rugvip — APPROVED** [review]

**Comment by freben** [comment]
I see, great.

**Pull request merged**